### ISSUE TITLE
Lots of random "red dots" showing up in the gutter.

### ISSUE THREAD

**Issue by Mario-M** [ggordan/GutterColor#93]
Noticed recently that there are a lot of red color circles showing up in my gutter where there is no red color defined in the adjacent line of code. Even happens on blank lines for that matter.

Any ideas?

![](http://i.imgur.com/otMwoFu.png)

**Comment by miguelolivares** [comment]
It happens because the "red" word inside "featured". Red is a color.

**Comment by jbrooksuk** [comment]
> It happens because the "red" word inside "featured". Red is a color.

I don't think that's true, the regex shouldn't be picking that up?

**Comment by Mario-M** [comment]
Yeah I don't think it's true either. That logic doesn't explain why they are showing up on brackets, white, white space, and comment lines, etc.

**Comment by miguelolivares** [comment]
Look at this: 
Any css color name (http://www.w3schools.com/cssref/css_colornames.asp) used in any part of a CSS document is taken as a color declaration. 

![screen shot 2015-10-29 at 0 32 39](https://cloud.githubusercontent.com/assets/4684028/10809376/d37b2c46-7dd4-11e5-8a63-b9e2780d7d72.png)

**Comment by miguelolivares** [comment]
My bad. I didn't see the red dot in empty lines.

**Comment by jbrooksuk** [comment]
That'll be the web colors highlighting. https://github.com/ggordan/GutterColor/blob/master/line.py#L207

**Comment by nbonamy** [comment]
Probably a colon is missing here but cannot figure where!

**Comment by digeomel** [comment]
I get the same. And it's not only red, it's seemingly random (or false) colours everywhere. There's obviously something wrong with the regex used (assuming it's a regex).
![screenshot from 2016-07-29 15-34-32](https://cloud.githubusercontent.com/assets/3295889/17249857/2583d826-55a2-11e6-96b3-c3a18fba3262.png)

**Comment by digeomel** [comment]
Hah, and to answer my own comment, it seem to pick up the "tan" colour from "important" :smile:

**Comment by digeomel** [comment]
Ok, this seems to work for me (from line.py, lines 204-214):

    def generate_webcolors(self):
    """Generates a list of web color names."""
    self.WEB_COLORS = dict((name, color) for (name, color) in css3_names_to_hex.items())
    self.WEB_COLORS_REGEX = '\\s(' + '|'.join(self.WEB_COLORS.keys()) + ')[\\s;]'
    # self.WEB_COLORS_REGEX = '((?<!\$)'+ '|(?<!\$)'.join(self.WEB_COLORS.keys()) +')'

    def web_color(self):
    """Returns the color in the line, if any CSS color name is found."""
    matches = re.search(self.WEB_COLORS_REGEX, self.text)
    if matches:
      return matches.group(1)

Note that because the regular expression is changed to match blank characters at the start and end of the named colour, you have to change line 214 as well to read `matches.group(1)` instead of `matches.group(0)`

**Comment by digeomel** [comment]
In retrospect the regex can be improved a little bit more like this:

`self.WEB_COLORS_REGEX = '[\\s:](' + '|'.join(self.WEB_COLORS.keys()) + ')[\\s;]'`

But by now you get the point :smile:

**Comment by sAs59** [comment]
@digeomel it doesn't work without semicolon. In *.sass for example

**Comment by glennabaron** [comment]
Its been a minute, and its kind of hacky, but in case this helps anyone else: If you try to stay away from using color names (and mainly use HEX or RGBA, etc.), just remove the list of web safe colors 👍
